src/sentry/sentry_metrics/indexer/limiters/writes.py

```diff
@@ -33,22 +33,30 @@ def _build_quota_key(namespace: str, org_id: Optional[OrgId] = None) -> str:
 
 
 @metrics.wraps("sentry_metrics.indexer.construct_quotas")
-def _construct_quotas(use_case_id: UseCaseKey, namespace: str) -> Sequence[Quota]:
+def _construct_quotas(use_case_id: UseCaseID, namespace: str) -> Sequence[Quota]:
     """
     Construct write limit's quotas based on current sentry options.
 
     This value can potentially cached globally as long as it is invalidated
     when sentry.options are.
     """
-    if use_case_id == UseCaseKey.PERFORMANCE:
+    if use_case_id is UseCaseID.TRANSACTIONS:
         return [
             Quota(prefix_override=_build_quota_key(namespace, None), **args)
```

Comment:
I think one simplification that is possible is to not pass `namespace` to `_construct_quotas` and instead build the quota keys with the use_case_Id string instead. I think this would work fine as these keys are for identifying the specific quotas.

Reply:
I'm a little hesitant to apply this change.

It seems like`RedisSlidingWindowRateLimiter` is using this `_build_quota_key` function (which makes use of `namespace`) to determine if it wants to override global / per-organization limits.

I'm not really familiar with the `RedisSlidingWindowRateLimiter` so I'm not certain whether or not we can just replace the namespace with our use case ID or if we also need to change the configuration elsewhere when we do this.

Reply:
Hmm let me check on this...also @untitaker might have an idea on whether this would be okay?

Reply:
Added the use case ID to _get_quota_key